File: typescript-examples/src/data-expiry-examples.ts
import { DataExpiryApi, ServerConfiguration, createConfiguration } from "@privatedataservices/metronome";

const clientConfig = createConfiguration( {
    baseServer: new ServerConfiguration("pds-internal-code-examples", "eu-north-1", {}, false),
    authMethods: { ApiKeyAuth: process.env.METRONOME_API_KEY }
});
const expiryApiClient = new DataExpiryApi(clientConfig);

const queryExpiriesExample = async () => {
    try {
        const response = await expiryApiClient.getExpiryNotices("20230809");
        
        // Pending expiries may change with use before the currently reported date.  They are advisory only.
        response.pending.forEach( (notice) => {
            if(notice.itemId) {
                console.log("Item %s expires on %s", notice.itemId, response.expiryDate);
            } else {
                console.log("Item %s has sub-items %s expirying on %s", notice.parentItemId, JSON.stringify(notice.subItems), response.expiryDate);
            }
        });

        // Complete expiries indicate that the data must be deleted.  You must act on complete expiry notices.
        response.complete.forEach( (notice) => {
            if(notice.itemId) {
                console.log("Item %s did expire on %s", notice.itemId, response.expiryDate);
            } else {
                console.log("Item %s has sub-items %s that did expire on %s", notice.parentItemId, JSON.stringify(notice.subItems), response.expiryDate);
            }
        });
    } catch (err) {
        console.error("Failed to expiry notices because: %s", JSON.stringify(err));
    }
}

File: ruby-examples/data_expiry_examples.rb
require 'metronome'

def query_expiries_example(client)
    begin
        response = client.get_expiry_notices(Date.parse("2025-05-12"))

        # Pending expiries may change with use before the currently reported date.  They are advisory only.
        puts "There are #{response.pending.length} pending expiries."
        response.pending.each do |notice|
            if notice.is_a?(Metronome::ItemExpiry)
                puts "Item #{notice.item_id} will currently expire on #{response.expiry_date}"
            else
                puts "Item #{notice.parent_item_id} sub-items #{notice.sub_items} will currently expire on #{response.expiry_date}"
            end
        end

        # Complete expiries indicate that the data must be deleted.  You must act on complete expiry notices.
        puts "There are #{response.complete.length} complete expiries."
        response.complete.each do |notice|
            if notice.is_a?(Metronome::ItemExpiry)
                puts "Item #{notice.item_id} did expire on #{response.expiry_date}"
            else
                puts "Item #{notice.parent_item_id} sub-items #{notice.sub_items} did expire on #{response.expiry_date}"
            end
        end
    rescue Metronome::ApiError => e
    puts "Exception when calling DataExpiryApi->get_expiry_notices: #{e}"
    end
end

# Configure the client
Metronome.configure do |config|
    config.api_key['ApiKeyAuth'] = ENV["METRONOME_API_KEY"]
    config.customer_id = "pds-internal-code-examples"
    config.region = "eu-north-1"
end

expiry_api_client = Metronome::DataExpiryApi.new

query_expiries_example(expiry_api_client)
